### PR TITLE
LPS-164617 | Add Autom. test FDSCustomView#CanBeEdited

### PR DESCRIPTION
**Pull request by mateussandes23**
**Background**

Create automation tests for Views

**Test Plan**

- Create a new view
- Make some changes on this view
- Save the changes
- Assert if the changes on new view were applyed

**JIRA Ticket:**
https://issues.liferay.com/browse/LPS-164617

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path [hunk 1]
```diff
@@ -209,6 +214,30 @@
 <tr>
 	<td>SELECT_OPTION_CUSTOM_VIEW</td>
 	<td>//button[contains(@class,'dropdown-item') and contains(text(), '${key_itemName}')]</td>
+</tr>
+<tr>
+	<td>SELECT_OPTION</td>
+	<td>//button[contains(@class,'dropdown-item') and contains(text(), '${key_optionName}')]</td>
+	<td></td>
```
Comment (mateusralv) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path:221]: This path is the same `SELECT_OPTION_CUSTOM_VIEW`

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path [hunk 2]
```diff
@@ -209,6 +214,30 @@
...
+</tr>
+<tr>
+	<td>INPUT_CUSTOM_VIEW_NAME</td>
+	<td>//input[contains(@id,'LabelInput')]</td>
```
Comment (mateusralv) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path:225]: This path is duplicated

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path [hunk 3]
```diff
@@ -209,6 +214,30 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>TABLE_CELL</td>
+	<td>xpath=(//div[@class='dnd-td' and position()='${key_columnIndex}'])['${key_valueIndex}']</td>
+	<td></td>
+</tr>
```
Comment (mateusralv) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path:232]: Same this path

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path [hunk 4]
```diff
@@ -209,6 +214,30 @@
...
+<tr>
+	<td>SAVE_BUTTON</td>
+	<td>//button[contains(text(), 'Save View')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>SELECT_OPTION</td>
+	<td>//button[contains(@class,'dropdown-item') and contains(text(), '${key_optionName}')]</td>
```
Comment (mateusralv) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path:240]: Same here.

**Comment by mateusralv** [comment]
ci:test:frontend:nocompile

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path [hunk 5]
```diff
@@ -221,6 +226,16 @@
 	<td>//button[contains(@class,'dropdown-item') and contains(text(), '${key_itemName}')]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>INPUT_CUSTOM_VIEW_NAME2</td>
+	<td>//input[contains(@id,'LabelInput')]</td>
+	<td></td>
```
Comment (mateusralv) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path:232]: This path is the same as `INPUT_CUSTOM_VIEW_NAME` can you change this one?

**Comment by john-co** [comment]
Just started reviewing :)

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase [hunk 6]
```diff
@@ -48,7 +48,39 @@ definition {
...
+		}
+
+		task ("And Click Save button") {
+			FDSCustomView.saveCustomView();
+		}
+
+		task ("Change selection to default view and assert an aspect from default view") {
+			FDSCustomView.viewNoColumn(key_field = "Description");
+
+			FDSCustomView.verifyFilterSummaryLabel(key_filter = "Blue, Green, Yellow");
```
Comment (john-co) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase:72]: This assertion should be "Green, Yellow, Red"

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase [hunk 7]
```diff
@@ -48,7 +48,39 @@ definition {
...
+		}
+
+		task ("then change selection to editted custom view and assert some aspects of it ") {
+			Click(locator1 = "FrontendDataSet#SELECT_CUSTOM_VIEW");
+
+			FDSCustomView.selectView(key_itemName = "Default View");
+
+			FDSCustomView.viewColumn(key_field = "Description");
+
+			FDSCustomView.verifyFilterSummaryLabel(key_filter = "Green, Yellow, Red");
```
Comment (john-co) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase:82]: This assertion should be "Blue, Green, Yellow". The reason why the test passed is b/c VerifyElementPresent does not fail the test if the element is not present. It will continue with the test and only highlight the step as a warning.

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path [hunk 8]
```diff
@@ -211,6 +211,11 @@
 
 <!--CUSTOM VIEWS-->
 
+<tr>
+	<td>OPEN_SELECT_FIELD</td>
+	<td>//button[contains(@aria-label, "Open Fields Menu")]</td>
+	<td></td>
+</tr>
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path:218]: none of these 3 paths are used in the new test, why are these necessary?

**Review by john-co — COMMENTED** [review]
refactored and applied suggestions to https://github.com/liferay-frontend/liferay-portal/pull/2854
